setup.py

```diff
@@ -21,13 +21,7 @@ def read(*names, **kwargs):
 
 
 extra_require = {
-    'dev': [
-        'grpcio_tools',
-        'mypy>=0.701',
-        'mypy-protobuf',
-        'pillow',
-        'matplotlib',
-    ],
+    'plugins': ['hangar_pil', 'hangar_matplotlib'],
```

Comment:
These should not be requirements of the hangar project itself. If the user wants them, they should download them. If we do this, it means every installation of hangar will download PIL, Matplotlib, and all of their dependencies. It's a lot of downloading to do if you don't need it.

Reply:
But it's an extra_require and will be installed only if user do `pip install hangar[all]` or `pip install hangar[plugins]`. And we needed to give hangar as **batteries included** anyways. What do you think?

Reply:
I disagree. The ability to do a `pip install[foo]` is as hidden a feature as any in the crazy world of python packaging. I didn't even know it was a thing until fairly recently. If we're going to write documentation on how to install and use these plugins, it's just easier to write the same "how-to" documentation and say to just treat it like any normal python plugin package:

    $ pip install hangar_matplotlib hangar_pil

More importantly than anything, if we include these as an option in the setup script, we need to test them fully. Any `pip install hangar` should be fully covered by our testing and performance guarrentees. We wouldn't be treating them as a "plugin", but as something "we plug in to". I agree that we should be "batteries included", but we are "batteries included" for the things which are relevant to Hangar: efficiently storing, retrieving, accessing, sharing, and collaborating on huge sets numerical data. The reason this project even works is because we don't care an ounce about the domain specific meanings that some set of stored bytes has. We just care about how we can most effectively get it in and out. 

I don't want to have to care about new releases of `matplotlib` or `pil` when I go to push a release, it's irrelevant to us, and a risk to include them as even something which could remotely be construed as "something the core team guarantees/supports".